sklearn/cluster/tests/test_k_means.py

```diff
@@ -1030,3 +1031,23 @@ def test_minibatch_kmeans_wrong_params(param, match):
     # are passed for the MiniBatchKMeans specific parameters
     with pytest.raises(ValueError, match=match):
         MiniBatchKMeans(**param).fit(X)
+
+
+@pytest.mark.parametrize("Estimator", [KMeans, MiniBatchKMeans])
+def test_mkl_vcomp_warning(Estimator):
+    # Check the memory leak warning.
+    modules = threadpool_info()
+    has_vcomp = "vcomp" in [module["prefix"] for module in modules]
+    has_mkl = ("mkl", "intel") in [
+        (module["internal_api"], module.get("threading_layer", None))
+        for module in modules]
+
+    if not (has_vcomp and has_mkl):
+        pytest.skip("Requires vcomp and MKL")
+
+    km = Estimator(n_clusters=n_clusters, random_state=0)
+    if Estimator is KMeans:
+        km.set_params(algorithm="full")
+
+    with pytest.warns(UserWarning, match="memory leak on Windows"):
```

Comment:
Shall we only make this check if the number of thread is problematic. I assume this is not the case on Azure currently>

Reply:
Or for this test, we need to mock the number of threads

Reply:
I suspect we only have one core on the windows ci so we can't trigger the warning at all.

Reply:
So as I suspected we only have 1 core on the windows ci. If we can't trigger the warning in our ci don't you think I should just remove the test ? (cc @NicolasHug)

Reply:
yeah let's remove then, thanks for trying